ecaudit/src/test/java/com/ericsson/bss/cassandra/ecaudit/auth/TestWrappingAuditAuthenticator.java

```diff
@@ -0,0 +1,147 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class TestWrappingAuditAuthenticator
+{
+    @Mock
+    IAuditAuthenticator mockAuthenticator;
+
+    @Mock
+    AuditAdapter mockAuditAdapter;
+
+    @Mock
+    AuditConfig mockConfig;
+
+    @Mock
+    IAuditAuthenticator.AuditSaslNegotiator mockSaslNegotiator;
+
+    @InjectMocks
+    WrappingAuditAuthenticator authenticator;
+
+    @BeforeClass
+    public static void beforeClass()
+    {
+        ClientInitializer.beforeClass();
+    }
+
+    @Before
+    public void setup()
+    {
+        when(mockAuthenticator.createAuditedSaslNegotiator()).thenReturn(mockSaslNegotiator);
+
+        authenticator.setup();
+        authenticator.protectedResources();
+        authenticator.requireAuthentication();
+        authenticator.validateConfiguration();
+        authenticator.legacyAuthenticate(Collections.emptyMap());
+
+        verify(mockAuthenticator, times(1)).setup();
+        verify(mockAuthenticator, times(1)).protectedResources();
+        verify(mockAuthenticator, times(1)).requireAuthentication();
+        verify(mockAuthenticator, times(1)).validateConfiguration();
+        verify(mockAuthenticator, times(1)).legacyAuthenticate(eq(Collections.emptyMap()));
+    }
+
+    @After
+    public void after()
+    {
+        verifyNoMoreInteractions(mockAuthenticator, mockAuditAdapter);
+    }
+
+    @AfterClass
+    public static void afterClass()
+    {
+        ClientInitializer.afterClass();
+    }
+
+    @Test
+    public void testWhenUserIsProviedAuthenticationAttemptIsLogged()
+    {
+        AuthenticatedUser expected = mock(AuthenticatedUser.class);
+        when(mockSaslNegotiator.getAuthenticatedUser()).thenReturn(expected);
+        when(mockSaslNegotiator.getUser()).thenReturn("audited_user");
+
+        IAuthenticator.SaslNegotiator negotiator = authenticator.newSaslNegotiator();
+        AuthenticatedUser actual = negotiator.getAuthenticatedUser();
+
+        assertThat(actual).isSameAs(expected);
+        verify(mockAuditAdapter, times(1)).auditAuth(eq("audited_user"), eq(Status.ATTEMPT), anyLong(), any(Optional.class));
+        verify(mockAuditAdapter, times(1)).auditAuth(eq("audited_user"), eq(Status.SUCCEEDED), anyLong(), any(Optional.class));
+    }
+
+    @Test
+    public void testWhenAuthenticationFailsIsStillAudited()
+    {
+        when(mockSaslNegotiator.getUser()).thenReturn("audited_user");
+        when(mockSaslNegotiator.getAuthenticatedUser()).thenThrow(new AuthenticationException("audited_user"));
+
+        IAuthenticator.SaslNegotiator negotiator = authenticator.newSaslNegotiator();
+        try
+        {
+            negotiator.getAuthenticatedUser();
+            fail("Expected exception!");
+        }
+        catch (Exception e)
+        {
```

Comment:
you could use some nice Assertions stuff for exceptions, for example:
assertThatExceptionOfType(<Exception>.class)
        .isThrownBy(() -> negotiator.getAuthenticatedUser())
        .withMessageContaining("<If you want to assert on that...>");

Reply:
Thanks, I was trying to do this but didn't get it to work. I think I used assertThat wrong.